src/snowflake/cli/_plugins/workspace/manager.py

```diff
@@ -0,0 +1,40 @@
+from pathlib import Path
+from typing import Dict
+
+from snowflake.cli.api.entities.common import EntityActions
+from snowflake.cli.api.exceptions import InvalidProjectDefinitionVersionError
+from snowflake.cli.api.project.schemas.entities.entities import (
+    Entity,
+    v2_entity_model_to_entity_map,
+)
+from snowflake.cli.api.project.schemas.project_definition import DefinitionV20
+
+
+class WorkspaceManager:
+    """
+    Instantiates entity instances from entity models, providing higher-order functionality on entity compositions.
+    """
+
+    def __init__(self, project_definition: DefinitionV20, project_root: Path):
```

Comment:
Since __init__ is checking if version is V2, it means users can pass in non-V2 PDF. I suggest we change the signature to ProjectDefinition then.